homeassistant/components/sensor/arlo.py

```diff
@@ -133,18 +163,31 @@ def update(self):
             except TypeError:
                 self._state = None
 
+        elif self._sensor_type == 'temperature':
+            try:
+                self._state = self._data.ambient_temperature
+            except TypeError:
+                self._state = None
+
+        elif self._sensor_type == 'humidity':
+            try:
+                self._state = self._data.ambient_humidity
+            except TypeError:
+                self._state = None
+
+        elif self._sensor_type == 'air_quality':
+            try:
+                self._state = self._data.ambient_air_quality
+            except TypeError:
+                self._state = None
+
     @property
     def device_state_attributes(self):
         """Return the device state attributes."""
         attrs = {}
 
         attrs[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
         attrs['brand'] = DEFAULT_BRAND
-
-        if self._sensor_type == 'last_capture' or \
-           self._sensor_type == 'captured_today' or \
-           self._sensor_type == 'battery_level' or \
-           self._sensor_type == 'signal_strength':
-            attrs['model'] = self._data.model_id
+        attrs['model'] = self._data.model_id
```

Comment:
Add a test or update the test where all sensor types are set up, and test that attributes return ok also for the total cameras type. I think we need to add a check here where we don't add the model attribute for that sensor type.